src/reanimated2/hook/useAnimatedRef.ts

```diff
@@ -6,11 +6,20 @@ import { getTag } from '../NativeMethods';
 import { getShadowNodeWrapperFromHostInstance } from '../fabricUtils';
 import {
   makeShareableCloneRecursive,
+  makeShareableShadowNodeWrapper,
   registerShareableMapping,
 } from '../shareables';
 
+function getShareableShadowNodeFromComponent(
+  component: Component
+): ShadowNodeWrapper {
```

Comment:
It looks like this function has an invalid return type. Technically, it returns a shareable, not a ShadowNodeWrapper directly.

There's no error since we explicitely define `ShadowNodeWrapper` type as `object` here:

 https://github.com/software-mansion/react-native-reanimated/blob/99573c396ca6b7e05ec28b5f2165ee7ad06c6dc5/src/reanimated2/commonTypes.ts#L174

Reply:
I don't understand what the problem is here? `makeShareableShadowNodeWrapper` return type is the same as the argument and in this case it's `ShadowNodeWrapper`. There would be no error even if `ShadowNodeWrapper` was defined differently

Reply:
Indeed, `makeShareableShadowNodeWrapper` returns the argument ([proof](https://github.com/software-mansion/react-native-reanimated/blob/99573c396ca6b7e05ec28b5f2165ee7ad06c6dc5/src/reanimated2/shareables.ts#L29-L34)). Everything is okay.